lxd/db/backups.go

```diff
@@ -541,3 +552,185 @@ func (c *ClusterTx) RenameVolumeBackup(ctx context.Context, oldName, newName str
 
 	return nil
 }
+
+// GetStoragePoolBucketBackupsName returns the names of all backups of the storage bucket with the given name.
+func (c *ClusterTx) GetStoragePoolBucketBackupsName(ctx context.Context, projectName string, bucketName string) ([]string, error) {
+	var result []string
+
+	q := `SELECT storage_buckets_backups.name FROM storage_buckets_backups
+JOIN storage_buckets ON storage_buckets_backups.storage_bucket_id=storage_buckets.id
+JOIN projects ON projects.id=storage_buckets.project_id
+WHERE projects.name=? AND storage_buckets.name=?
+ORDER BY storage_buckets_backups.id`
+	inargs := []any{projectName, bucketName}
+	outfmt := []any{bucketName}
+
+	dbResults, err := queryScan(ctx, c, q, inargs, outfmt)
+	if err != nil {
+		return nil, err
+	}
+
+	for _, r := range dbResults {
+		result = append(result, r[0].(string))
+	}
+
+	return result, nil
+}
+
+// CreateStoragePoolBucketBackup creates a new storage bucket backup.
+func (c *ClusterTx) CreateStoragePoolBucketBackup(ctx context.Context, args StoragePoolBucketBackup) error {
+	_, err := c.getStoragePoolBucketBackupID(ctx, args.Name)
+	if err == nil {
+		return ErrAlreadyDefined
```

Comment:
we should use api.StatusErrorf() here like we do in `CreateStoragePoolBucketKey` then we wont need the new ErrAlreadyDefined var. 

```go
 api.StatusErrorf(http.StatusConflict, "A bucket key using that access key already exists on this server")
```

Then we can check it in the call site using:

```
api.StatusErrorCheck(err, http.StatusConflict)
```